akka-actor-tests/src/test/scala/akka/pattern/ActorExecutionContextSpec.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.pattern
+
+import java.util.concurrent.ThreadLocalRandom
+
+import akka.actor.Actor
+import akka.actor.NoSerializationVerificationNeeded
+import akka.actor.Props
+import akka.testkit.AkkaSpec
+import akka.testkit.TestProbe
+
+import scala.concurrent.duration._
+import scala.concurrent.Future
+
+class ActorExecutionContextSpec extends AkkaSpec {
+  "The ActorExecutionContext pattern" must {
+    "run thunks inside the actor context" in {
+      case class Inc(by: Int) extends NoSerializationVerificationNeeded
+
+      class ValueActor extends Actor with ActorExecutionContext {
+        private[this] var _counter: Int = 0
+
+        def receive: Receive = {
+          case "get" ⇒ sender() ! _counter
+          case Inc(by) ⇒
+            akka.pattern.after(ThreadLocalRandom.current().nextInt(10).millis, system.scheduler) {
+              Future(_counter += by)
```

Comment:
"Concurrent"-looking updates of the `_counter` var. But these are actually safe.